### PR TITLE
[tinyfiledialogs] Update version and download path to v3.6.3

### LINKED ISSUE #12136: [tinyfiledialogs] update to 3.6.3
Library name: tinyfiledialogs

New version number: 3.6.3

Other information that may be useful (release notes, etc...): includes a fixed link that will never be tampered with.

### PR DESCRIPTION
**Pull request by MaxKruse**
This PR will increase the version of tinyfiledialogs to 3.6.3, as well as provide a frozen_version link from the author itself, guaranteeing the possibility to switch versions without much hassle

- What does your PR fix? Fixes #12136

- Which triplets are supported/not supported? Have you updated the CI baseline? edited

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? I tried my best, so i hope it does :P

### CODE REVIEW

File: ports/tinyfiledialogs/portfile.cmake [hunk 1]
```diff
@@ -5,9 +5,8 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_sourceforge(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO tinyfiledialogs
-    REF v3.4
-    FILENAME "tinyfiledialogs-3.4.3.zip"
-    SHA512 b4a8c8fa5ff53a0972ce9dd1a4a473eaeb82689e5a47553b83e9220ea7e0ec582d87111728088ab6d314972e6531653b11fbd8c05c5e46fbe5bc4d29c7fe23fb
+    FILENAME "frozen_versions/tinyfiledialogs-3.6.3.zip"
+    SHA512 42c3bd34b0287cf2477f9ede049bea29a9306304e8fab7740065957d3737f4041899f26f29a0693e801cb0a7b63844509f86441262303ff0a4030a431ffac648
 )
```
Comment (NancyLi1013) [inline ports/tinyfiledialogs/portfile.cmake:11]: Could you please use `vcpkg_replace_string()` to handle the strings that need to replace in a file?

Comment (MaxKruse) [reply ports/tinyfiledialogs/portfile.cmake:11]: I am unsure about 2 things here:

Firstly, I am not sure if replacing a string IN the file is needed at all, what needs to be changed is the filename itself, not its content.
Secondly, due to my own lack of knowledge regarding CMake, i was unable to make a simple "replace the version number" in the download string, as well as the downloadpath prefix

Comment (NancyLi1013) [reply ports/tinyfiledialogs/portfile.cmake:11]: Thanks for your clarification about this. In this case, it seems that there is no need to replace.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the update!

**Pull request merged**